tests/integration/pilot/mcs/discoverability/discoverability_test.go

```diff
@@ -234,13 +236,77 @@ func checkClustersReached(t framework.TestContext, src, dest echo.Instance, clus
 		dest.Config().Service,
 		dest.Config().Namespace.Name())
 
-	src.CallWithRetryOrFail(t, echo.CallOptions{
+	_, err := src.CallWithRetry(echo.CallOptions{
 		Address:   address,
 		Target:    dest,
 		Count:     50,
 		PortName:  "http",
 		Validator: echo.And(echo.ExpectOK(), echo.ExpectReachedClusters(clusters)),
 	}, retryTimeout)
+	if err != nil {
+		t.Fatal(fmt.Errorf("%v\nCluster Details:\n%s", err, getClusterDetailsYAML(t, src, dest)))
```

Comment:
nit: would you also want this on success to have something to compare against?

Reply:
I don't think we'd need it, honestly. But we can change that in the future if we think it will be helpful.